docs/index.md

```diff
@@ -1,11 +1,54 @@
-# Github-runner operator
-A [Juju](https://juju.is/) [charm](https://juju.is/docs/olm/charmed-operators) deploying self-hosted GitHub runners.
-  
-Each unit of this charm will start a configurable number of LXD based containers and virtual machines to host them. Each runner performs only one
-job, after which it unregisters from GitHub to ensure that each job runs in a clean environment. The charm will periodically check the number of idle runners and spawn or destroy them as
-necessary to maintain the configured number of runners. Both the reconciliation interval and the number of runners to maintain are configurable.
+A [Juju](https://juju.is/) [charm](https://juju.is/docs/olm/charmed-operators) deploying and managing [GitHub self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) on virtual machines.
+
+This charm simplifies initial deployment and "day N" operations of GitHub self-hosted runners. The charm makes it easy to manage self-hosted runners with security and hardware resource usage in mind.
+
+Operating your own GitHub self-hosted runner comes with [its own security concerns according to GitHub](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security).
+Just like GitHub, the self-hosted runner managed by the charm are isolated in a single-use virtual machine. However, arbitrary code execution is possible under certain repositories settings. This can be utilized by malicious actors in a number of ways, such as, crypto-mining. To combat this, the charm enforce a set of GitHub repository settings to ensure code that executed in the self-hosted runner are reviewed by someone trusted.
+
+The charm also upgrades dependencies on a schedule to migrate security risks. The upgrades includes linux kernel upgrades and would automated the rebooting process. This ensures the latest security patches are installed within mintues.
```

Comment:
s/migrate/mitigate
s/The upgrades includes linux kernel upgrades and would automated the rebooting process/The upgrade includes linux kernel upgrades,  automatically rebooting the machines